src/framework/audio/iplayback.h

```diff
@@ -19,36 +19,35 @@
  * You should have received a copy of the GNU General Public License
  * along with this program.  If not, see <https://www.gnu.org/licenses/>.
  */
-#ifndef MU_AUDIO_IAUDIOENGINE_H
-#define MU_AUDIO_IAUDIOENGINE_H
-
-#include <string>
+#ifndef MU_AUDIO_ISEQUENCER_H
+#define MU_AUDIO_ISEQUENCER_H
 
 #include "modularity/imoduleexport.h"
-
-#include "ret.h"
 #include "async/channel.h"
-#include "isequencer.h"
-#include "iaudiodriver.h"
-#include "internal/iaudiobuffer.h"
-#include "imixer.h"
-#include "isynthesizer.h"
+#include "async/promise.h"
+
+#include "iplayers.h"
+#include "itracks.h"
+#include "iaudioio.h"
+#include "audiotypes.h"
 
 namespace mu::audio {
-class IAudioEngine : MODULE_EXPORT_INTERFACE
+class IPlayback : MODULE_EXPORT_INTERFACE
 {
-    INTERFACE_ID(IAudioEngine)
+    INTERFACE_ID(IPlayback)
 
 public:
-    virtual ~IAudioEngine() = default;
+    virtual ~IPlayback() = default;
 
-    virtual bool isInited() const = 0;
-    virtual async::Channel<bool> initChanged() const = 0;
-    virtual unsigned int sampleRate() const = 0;
-    virtual ISequencerPtr sequencer() const = 0;
-    virtual IMixerPtr mixer() const = 0;
-    virtual IAudioBufferPtr buffer() const = 0;
-    virtual void setAudioBuffer(IAudioBufferPtr) = 0;
+    virtual async::Promise<TrackSequenceId> addSequence() = 0;
+    virtual async::Promise<TrackSequenceIdList> sequenceIdList() const = 0;
+    virtual void removeSequence(const TrackSequenceId id) = 0;
+
+    virtual ITracksPtr tracks() const = 0;
+    virtual IPlayersPtr players() const = 0;
+    virtual IAudioIOPtr audioIO() const = 0;
 };
```

Comment:
I think needs comments for this interface, like:
```
    // 1. Add Sequence
    virtual async::Promise<TrackSequenceId> addSequence() = 0;
    virtual async::Promise<TrackSequenceIdList> sequenceIdList() const = 0;
    virtual void removeSequence(const TrackSequenceId id) = 0;

    // 2. Setup tracks for Sequence
    virtual ITracksPtr tracks() const = 0;

    // 3. Play Sequence
    virtual IPlayersPtr players() const = 0;

    // 4. Output params
    virtual IAudioIOPtr audioIO() const = 0;
```

Reply:
yep, done